backend/handlers/bets/betutils/feeutils_test.go

```diff
@@ -0,0 +1,198 @@
+		t.Fatalf("Failed to connect to the database: %v", err)
+	}
+
+	// Migrate the Bet and User models
+	db.AutoMigrate(&models.Bet{}, &models.User{})
+
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Create a test user with a unique api_key
+	user := &models.User{
+		Username:       "testuser",
+		AccountBalance: 1000,
+		ApiKey:         "unique_api_key_1", // Ensure this is unique
+	}
+	db.Create(&user) // Save the user to the database
+
+	// Scenario 1: User places a bet on Market 1 where they have no prior bets
+	initialBetFee := getUserInitialBetFee(db, 1, user)
+	if initialBetFee != appConfig.Economics.Betting.BetFees.InitialBetFee {
+		t.Errorf("Expected initial bet fee to be %d, got %d", appConfig.Economics.Betting.BetFees.InitialBetFee, initialBetFee)
+	}
+
+	// Place a bet for the user on Market 1
+	bets := []models.Bet{
+		{Username: "testuser", MarketID: 1, Amount: 100, PlacedAt: time.Now()},
+	}
+	db.Create(&bets) // Save the bet to the database
+
+	// Scenario 2: User places another bet on Market 1 where they already have a bet
+	initialBetFee = getUserInitialBetFee(db, 1, user)
+	if initialBetFee != 0 {
+		t.Errorf("Expected initial bet fee to be 0, got %d", initialBetFee)
+	}
+
+	// Scenario 3: User places a bet on Market 2 where they have no prior bets
+	initialBetFee = getUserInitialBetFee(db, 2, user)
+	if initialBetFee != appConfig.Economics.Betting.BetFees.InitialBetFee {
+		t.Errorf("Expected initial bet fee to be %d, got %d", appConfig.Economics.Betting.BetFees.InitialBetFee, initialBetFee)
+	}
+
+}
+
+func TestGetTransactionFee(t *testing.T) {
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Test buy scenario
+	buyBet := models.Bet{Amount: 100}
+	transactionFee := getTransactionFee(buyBet)
+	if transactionFee != appConfig.Economics.Betting.BetFees.BuySharesFee {
+		t.Errorf("Expected buy transaction fee to be %d, got %d", appConfig.Economics.Betting.BetFees.BuySharesFee, transactionFee)
+	}
+
+	// Test sell scenario
+	sellBet := models.Bet{Amount: -100}
+	transactionFee = getTransactionFee(sellBet)
+	if transactionFee != appConfig.Economics.Betting.BetFees.SellSharesFee {
+		t.Errorf("Expected sell transaction fee to be %d, got %d", appConfig.Economics.Betting.BetFees.SellSharesFee, transactionFee)
+	}
+}
+
+func TestGetSumBetFees(t *testing.T) {
+	// Set up in-memory SQLite database
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	if err != nil {
+		t.Fatalf("Failed to connect to the database: %v", err)
+	}
+
+	// Migrate the Bet model
+	db.AutoMigrate(&models.Bet{})
+
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Create a test user
+	user := &models.User{Username: "testuser"}
+
+	// Scenario 1: User has no bets, buys shares, gets initial fee
+	buyBet := models.Bet{MarketID: 1, Amount: 100}
+	sumOfBetFees := GetSumBetFees(db, user, buyBet)
+	expectedSum := appConfig.Economics.Betting.BetFees.InitialBetFee
```

Comment:
Is the initial bet in addition to or instead of the respective buy/sell fees? If it's in addition to, I think we need to add the buy fee here.